targets/mbedos5/jerryscript-mbed/jerryscript-mbed-drivers/source/DigitalOut.cpp

```diff
@@ -0,0 +1,139 @@
+/* Copyright (c) 2016 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "jerryscript-mbed-util/logging.h"
+#include "jerryscript-mbed-library-registry/wrap_tools.h"
+
+#include "mbed.h"
+
+/**
+ * DigitalOut#write (native JavaScript method)
+ *
+ * Writes a binary value to a DigitalOut.
+ *
+ * @param value 1 or 0, specifying whether the output pin is high or low,
+ *      respectively
+ * @returns undefined, or an error if invalid arguments are provided.
+ */
+DECLARE_CLASS_FUNCTION(DigitalOut, write) {
+    // Extract native DigitalOut pointer
+    uintptr_t ptr_val;
+    jerry_get_object_native_handle(this_obj, &ptr_val);
+
+    DigitalOut* native_ptr = reinterpret_cast<DigitalOut*>(ptr_val);
+
+    if (args_count == 1 && jerry_value_is_number(args[0])) {
+        int arg0 = jerry_get_number_value(args[0]);
+        native_ptr->write(arg0);
+
+        return jerry_create_undefined();
+    } else {
+        const char* error_msg = "Invalid arguments in method write";
+        return jerry_create_error(JERRY_ERROR_TYPE, reinterpret_cast<const jerry_char_t*>(error_msg));
```

Comment:
@akosthekiss you're right, the only reason it's like this is because DigitalOut.cpp was auto-generated using a script I wrote at some point, though for consistency since we're putting it into source control, I'll modify it to be consistent with the other driver sources.